### PR TITLE
Fixes #7608: Add a "quiet" option to rudder agent run to only display…

### PR DESCRIPTION
**Pull request by amousset**
… errors

https://www.rudder-project.org/redmine/issues/7608

### CODE REVIEW

**Comment by ncharles** [comment]
i don't have the knowledge to review this PR :/

**Comment by jooooooon** [comment]
Replaces #34

**Comment by jooooooon** [comment]
Thanks for setting this up Alexis. It looks technically correct, but I want to discuss the consequences and choices here a little more before we merge anything.

A few issues to consider:
  - Currently, if the agent run has no repairs nor errors, there is absolutely ZERO lines output from this command. Therefore, it is hard to tell whether it worked or not!
  - What about the rudder agent update and rudder agent inventory commands? They should behave the same way.
  - Should this not be the default? See @matya 's comment in the ticket on Redmine.


Any and all thoughts and comments welcome!

**Comment by jooooooon** [comment]
Asking for feedback from @peckpeck @ncharles @VinceMacBuche @Kegeruneku too

**Comment by peckpeck** [comment]
As said IRL: let's put the feature in 3.0, let's change the default behaviour in 3.2

**Comment by amousset** [comment]
PR updated

File: share/commands/agent-inventory [hunk 1]
```diff
@@ -4,5 +4,21 @@
...
+
+while getopts "q" opt; do
+  case $opt in
+    q)
+      FILTER="egrep -v @@result_success@@|@@result_na@@|@@log_info@@|@@log_debug@@|\[info\]|notice:"
+      ;;
+  esac
+done
+
+/var/rudder/cfengine-community/bin/cf-agent -KC -D force_inventory | ${FILTER}
```
Comment (jooooooon) [inline share/commands/agent-inventory]: Shouldn't this be -Calways ?

**Comment by amousset** [comment]
PR updated

File: share/commands/agent-inventory (hunk 1 above)
Comment (amousset) [inline share/commands/agent-inventory]: It should, updated.

**Comment by amousset** [comment]
PR updated

**Pull request merged**